circt: incompatible with cmake `find_package`

### Nixpkgs version

- Stable (24.11)

### Describe the bug

This nix package is unable to support `find_package` in CMake.

### Steps to reproduce

For your convenience, I have prepared an example project for reproducing this issue:

[circt-issue.tar.gz](https://github.com/user-attachments/files/19143012/circt-issue.tar.gz)

In brief, I just added one line, `find_package(CIRCT REQUIRED CONFIG)`, to a CMake hello world project. And I wrote the following nix derivation for it:
```nix
{
  stdenv,
  cmake,
  circt,
}:

stdenv.mkDerivation {
  pname = "hello";
  version = "0.1.0";
  src = ./.;

  nativeBuildInputs = [ cmake ];
  buildInputs = [ circt ];

  cmakeFlags = [
    "-DMLIR_DIR=${circt.passthru.llvm.dev}/lib/cmake/mlir"
  ];
}
```

And I ran `nix-build -A hello` to build this derivation.

### Expected behaviour

`find_package(CIRCT REQUIRED CONFIG)` is supported by this package.

### Screenshots

_No response_

### Relevant log output

```console
CMake Error at /nix/store/n9kky2n793wyj1zxwmk2qgj7f0mz8nv6-circt-1.87.0-dev/lib/cmake/circt/CIRCTTargets.cmake:1058 (message):
  The imported target "CIRCTDebugAnalysis" references the file

     "/nix/store/n9kky2n793wyj1zxwmk2qgj7f0mz8nv6-circt-1.87.0-dev/lib/libCIRCTDebugAnalysis.so.20.0git"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/nix/store/n9kky2n793wyj1zxwmk2qgj7f0mz8nv6-circt-1.87.0-dev/lib/cmake/circt/CIRCTTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /nix/store/n9kky2n793wyj1zxwmk2qgj7f0mz8nv6-circt-1.87.0-dev/lib/cmake/circt/CIRCTConfig.cmake:27 (include)
  CMakeLists.txt:4 (find_package)
```

### Additional context

As the error message suggested, CMake looked for `libCIRCTDebugAnalysis.so.20.0git` under `/nix/store/[...]-circt-1.87.0-dev/lib/`. However, its location would be `/nix/store/[...]-circt-1.87.0-lib/lib`.

After some investigation, I assumed this error was related to the following configurations in `/nix/store/[...]-circt-1.87.0-dev/lib/cmake/circt/CIRCTConfig.cmake`
```cmake
set(CIRCT_CMAKE_DIR "${CIRCT_INSTALL_PREFIX}/lib/cmake/circt")
set(CIRCT_BINARY_DIR "${CIRCT_INSTALL_PREFIX}")
set(CIRCT_INCLUDE_DIRS "${CIRCT_INSTALL_PREFIX}/include")
set(CIRCT_LIBRARY_DIRS "${CIRCT_INSTALL_PREFIX}/lib")
set(CIRCT_TOOLS_DIR "${CIRCT_INSTALL_PREFIX}/bin")
```
which conflicts with the real directory structure provided by `circt` in nixpkgs.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.74-1-MANJARO, Manjaro Linux, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.26.2`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/2nn2pqrb8kjy76kfnbh55rp1pwzml68i-nixpkgs/nixpkgs`

### Notify maintainers

@SharzyL @pineapplehunter @sequencer 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ci/circt/package.nix
+++ b/pkgs/by-name/ci/circt/package.nix
@@ -125,6 +125,17 @@ stdenv.mkDerivation rec {
 
   postInstall = ''
     moveToOutput lib "$lib"
+    moveToOutput lib/cmake "$dev"
+
+    substituteInPlace $dev/lib/cmake/circt/CIRCTConfig.cmake \
+      --replace-fail "\''${CIRCT_INSTALL_PREFIX}/lib/cmake/mlir" "${circt-llvm.dev}/lib/cmake/mlir" \
+      --replace-fail "\''${CIRCT_INSTALL_PREFIX}/lib/cmake/circt" "$dev/lib/cmake/circt" \
+      --replace-fail "\''${CIRCT_INSTALL_PREFIX}/include" "$dev/include" \
+      --replace-fail "\''${CIRCT_INSTALL_PREFIX}/lib" "$lib/lib" \
+      --replace-fail "\''${CIRCT_INSTALL_PREFIX}/bin" "$out/bin" \
+      --replace-fail "\''${CIRCT_INSTALL_PREFIX}" "$out"
+    substituteInPlace $dev/lib/cmake/circt/CIRCTTargets-release.cmake \
+      --replace-fail "\''${_IMPORT_PREFIX}/lib" "$lib/lib"
   '';
 
   passthru = {

